Libplanet/Blocks/Block.cs

```diff
@@ -229,31 +233,36 @@ private Block(RawBlock rb)
             return MakeBlock(nonce);
         }
 
-        public static Block<T> FromBencodex(byte[] encoded)
+        /// <summary>
+        /// Decodes a <see cref="Block{T}"/>'s
+        /// <a href="https://bencodex.org/">Bencodex</a> representation.
+        /// </summary>
+        /// <param name="bytes">A <a href="https://bencodex.org/">Bencodex</a>
+        /// representation of a <see cref="Block{T}"/>.</param>
+        /// <returns>A decoded <see cref="Block{T}"/> object.</returns>
+        /// <seealso cref="Serialize(bool, bool)"/>
+        public static Block<T> Deserialize(byte[] bytes)
         {
-            var serializer = new BencodexFormatter<Block<T>>();
-            using (var stream = new MemoryStream(encoded))
+            var value = new Codec().Decode(bytes);
+            if (!(value is Bencodex.Types.Dictionary dict))
             {
-                return (Block<T>)serializer.Deserialize(stream);
+                throw new DecodingException(
+                    $"Expected {typeof(Bencodex.Types.Dictionary)} but " +
+                    $"{value.GetType()}");
             }
+
+            return new Block<T>(dict);
         }
 
-        public byte[] ToBencodex(bool hash, bool transactionData)
+        public byte[] Serialize(bool hash, bool transactionData)
         {
-            var serializer = new BencodexFormatter<Block<T>>
-            {
-                Context = new StreamingContext(
-                    StreamingContextStates.All,
-                    new BlockSerializationContext(hash, transactionData)
-                ),
-            };
-            using (var stream = new MemoryStream())
-            {
-                serializer.Serialize(stream, this);
-                return stream.ToArray();
-            }
+            var codec = new Codec();
+            return codec.Encode(ToBencodex(hash, transactionData));
         }
 
+        public Bencodex.Types.Dictionary ToBencodex(bool hash, bool transactionData) =>
```

Comment:
You wrote, in the changelog, that this method was gone, but as the method of the same name still exists, it's better to be adjusted.